### ISSUE TITLE
Regression with webapp and dependency upgrade on track for 1.3 release

### ISSUE THREAD

**Issue by mitar** [meteor/meteor#6170]
(This was reported to https://github.com/peerlibrary/meteor-server-autorun/issues/10 by @huttonr.)

The commit https://github.com/meteor/meteor/commit/5d501d17936de234fa0e992e590f70bf958e98db introduced a regression.

Before, stack trace for calling `getBoilerplate` was:

```
Object.handle (packages/webapp/webapp_server.js:708:1)
next (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:190:15)
next (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:192:9)
next (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:127:23)
Function.app.handle (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:198:3)
Object.fn [as handle] (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:74:14)
next (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:190:15)
Object.WebAppInternals.staticFilesMiddleware (packages/webapp/webapp_server.js:362:1)
packages/webapp/webapp_server.js:634:1
```

After upgrade of dependencies it is:

```
at Object.handle (packages/webapp/webapp_server.js:714:1)
at next (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:174:15)
at Object.next (meteor/packages/webapp/.npm/package/node_modules/connect/lib/proto.js:176:9)
at Object.immediate._onImmediate (timers.js:381:16)
at processImmediate [as _immediateCallback] (timers.js:363:15)
```

Because `getBoilerplate` calls into Blaze and Blaze calls into Tracker, this is the issue for packages which try to provide server-side rendering and other server side capabilities (like server-side Tracker). Because now this [fiber wrapping here](https://github.com/meteor/meteor/blob/devel/packages/webapp/webapp_server.js#L627) is not in the effect anymore. It seems that internally `connect` or some other middleware now bounces serving of the file through the event loop, escaping the Fiber.

Because meteor was previously calling `getBoilerplate` inside a Fiber, I would claim that this now is a regression and probably goes against the intent why serving static files were wrapped into the Fiber in the first place.

It would be great if this could be resolved before the 1.3 release so that apps which use reactivity on the server side would not break. Currently you cannot load the app at all because getting boilerplate throws an exception.

cc @martijnwalraven

**Comment by martijnwalraven** [comment]
To be honest, I have no idea how upgrading `send` and `connect` causes this difference in behavior. You seem to understand the issue much better! Do you have suggestions for a fix?

**Comment by mitar** [comment]
I think it is this change: https://github.com/senchalabs/connect/commit/be0382481b684f6ef5893da0bb2247b0e177e488

**Comment by mitar** [comment]
Yea, it is nothing special. Just wrapping the function into a Fiber. I made a pull request for this: #6179. It is not changing anything except indenting once more (but diff is ugly).

**Comment by mitar** [comment]
But I must add that I am sad because this pull request will be probably merged sooner than my others for the same file which will then become invalid and I will have to do even more extra work to get them back to mergeable. This is really not nice. Because pull requests are being merged so slowly. Who has webapp over? @avital?

List of pull requests related to this file:
* #5714
* #6059
* #6060
* #6101

**Comment by ndarilek** [comment]
Any chance we'll see a new beta build soon with this fix incorporated? I'm stuck on beta6 because of this.

Thanks.

**Comment by martijnwalraven** [comment]
@ndarilek: These commits are already part of the Cordova 1.3 beta (see https://github.com/meteor/meteor/issues/6050). The `cordova-improvements` branch will be merged with `release-1.3` soon.
